src/screens/Ticketing/Purchase/Payment/SelectPaymentMethod/index.tsx

```diff
@@ -127,9 +132,22 @@ const SelectPaymentMethod: React.FC<Props> = ({
         },
       };
     });
-    return [...remoteOptions.reverse(), ...options];
+    return [...remoteOptions.reverse(), ...defaultPaymentOptions];
   }
 
+  const isSelectedOption = (item: SavedPaymentOption) =>
+    JSON.stringify({
+      type: selectedOption?.paymentType,
+      id:
+        !!selectedOption && isRecurring(selectedOption)
+          ? selectedOption.recurringPaymentId
+          : 0,
+    }) ===
+    JSON.stringify({
+      type: item.paymentType,
+      id: item.savedType === 'recurring' ? item.recurringCard.id : 0,
+    });
```

Comment:
Vet denne var her fra før, men synes det er litt "hackish" med å konvertere til strenger for sammenligning. Fungerer noe som dette?
```typescript
const isSelectedOption = (item: SavedPaymentOption) =>
    selectedOption?.paymentType === item.paymentType
    && selectedOption?.recurringPaymentId === item.recurringCard?.id
```

Reply:
Dette ser ut til å fungere, og skal ha identisk oppførsel som før, er det bedre?

```javascript
const isSelectedOption = (item: SavedPaymentOption) => {
  // False if types doesn't match
  if (!(selectedOption?.paymentType === item.paymentType)) return false;

  const itemIsReccurring = !!selectedOption && isRecurring(selectedOption);
  const selectedIsRecurring = item.savedType === 'recurring';

  // True if recurring ID matches
  if (itemIsReccurring && selectedIsRecurring) {
    return item.recurringCard.id === selectedOption.recurringPaymentId;
  }

  // True if both are not recurring, false otherwise
  return !itemIsReccurring && !selectedIsRecurring;
};
```

Reply:
Det er litt bedre ja. Føler fortsatt at det er i overkant komplisert i forhold til den enkle sjekken man skal gjøre 🤔  Noe av trøbbelet tror jeg kommer av at det er forskjellig typing på SavedPaymentOption og SelectedPaymentOption. Men kanskje ikke verd å investere mer tid i det.